src/main/scala/com/fulcrumgenomics/FgBioDef.scala

```diff
@@ -46,6 +46,17 @@ class FgBioDef extends CommonsDef {
     def bufferBetter = new BetterBufferedIterator(scala.collection.JavaConversions.asScalaIterator(iterator))
   }
 
+  /** Implicit that wraps a Java iterator into a Scala iterator. */
+  implicit class JavaIteratorAdapter[A](private val underlying: java.util.Iterator[A]) extends Iterator[A] {
```

Comment:
Could there be an implicit class that takes in a `java.util.Iterable` in its constructor and creates an `Iterator`?  Then we can replace `iterableAsScalaIterable` with `JavaIterableAsIterator` or the like.  This would mean that the `toIterator` is not right in your face, but I think the thing we want is iterable->iterator.